### DISCUSSION TITLE
while refer same name,which it refer to

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1497] [Q&A]
i have two same function name `AddNumbers` in my different files,their formal parameter type  is (INT,INT) and (STRING,INT)
i call this  function `AddNumbers`  at main.st .
it seems that it refer the first  function with param(INT,INT),
but another function with formal parameter type (STRING,INT) seems didn't work well.so can langium support multi refer same name?
![image](https://github.com/eclipse-langium/langium/assets/26995040/03e3cf0f-e9ad-4a42-8889-641de397c235)

**Comment by cdietrich** [comment]
you would need to adjust the scoping to look into the actual argument (types) and based on that put the correct function into the scope

**Reply by zrwsmd** [reply]
i override DefaultScopeProvider&getScope,and write a method inside it,but it seems didn't work out well
![image](https://github.com/eclipse-langium/langium/assets/26995040/6dedfa1b-1f5e-419d-acfd-c41acced2a2c)

**Reply by zrwsmd** [reply]
![38c47a32808323bed73df6b116d331b](https://github.com/eclipse-langium/langium/assets/26995040/b5500cd6-bf0c-4560-b4ec-6e663edbfb62)

**Reply by cdietrich** [reply]
why do you change the name?

i would expect your scoping to look like

- at the function call place get the arguments from the crrent place
- for all functions filter the correct ones
- return a scope with exactly these
- (duriung concent assist you may return the default one with all)

**Reply by zrwsmd** [reply]
so should i still need to i override DefaultScopeProvider&getScope?where should i write this steps you mentioned above

**Reply by cdietrich** [reply]
yes

**Reply by cdietrich** [reply]
how does your grammar for function calls look like

**Reply by zrwsmd** [reply]
StFunction:
    'FUNCTION' name=Identifier ('('  (reqParams+=ReqParams (',')?)* ')')? ':' variable_type=Native_Type_Name
     (((varInputs+=VarInput)|(varOutputs+=VarOutput)|(varLocals+=VarLocal))*
      (statementList+=Statement_list_single) )*
    'END_FUNCTION'
;

ReqParams:
 param=Identifier ':'  typeName=Native_Type_Name;

**Reply by msujew** [reply]
That's the grammar for function definitions. But Christian asked for the grammar rule you use to call those functions.
